build.ps1

```diff
@@ -70,162 +75,163 @@ function Invoke-Configure {
-    $env:PATH = New-PathString -StartingPath $env:PATH -Path 'C:\Program Files (x86)\MSBuild\14.0\bin\amd64;C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\amd64;C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\VCPackages;C:\WINDOWS\Microsoft.NET\Framework64\v4.0.30319;C:\WINDOWS\Microsoft.NET\Framework64\;C:\Program Files (x86)\Windows Kits\10\bin\x64;C:\Program Files (x86)\Windows Kits\10\bin\x86;C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.6.1 Tools\x64\'
+function Invoke-Lint([string]$ToolChain) {
+    rustup component add clippy
+    $clippyArgs += Convert-ArrayToArgs -arg A -list (Get-Content $PSScriptRoot\test\unexamined_lints.txt)
+    $clippyArgs += Convert-ArrayToArgs -arg A -list (Get-Content $PSScriptRoot\test\allowed_lints.txt)
+    $clippyArgs += Convert-ArrayToArgs -arg W -list (Get-Content $PSScriptRoot\test\lints_to_fix.txt)
+    $clippyArgs += Convert-ArrayToArgs -arg D -list (Get-Content $PSScriptRoot\test\denied_lints.txt)
+
+    $clippyCommand = "$(Get-CargoCommand $ToolChain) clippy --all-targets --tests -- $clippyArgs"
+    Write-Host "--- Running clippy!"
+    Write-Host "Clippy rules: $clippyCommand"
+    Invoke-Expression $clippyCommand
 }
 
-$env:PATH                       = New-PathString -StartingPath $env:PATH    -Path 'C:\Program Files\7-Zip'
-$env:PATH                       = New-PathString -StartingPath $env:PATH    -Path $ChocolateyHabitatBinDir
-$env:LIB                        = New-PathString -StartingPath $env:LIB     -Path $ChocolateyHabitatLibDir
-$env:INCLUDE                    = New-PathString -StartingPath $env:INCLUDE -Path $ChocolateyHabitatIncludeDir
-$env:SODIUM_LIB_DIR             = $ChocolateyHabitatLibDir
-$env:LIBARCHIVE_INCLUDE_DIR     = $ChocolateyHabitatIncludeDir
-$env:LIBARCHIVE_LIB_DIR         = $ChocolateyHabitatLibDir
-$env:OPENSSL_LIBS               = 'ssleay32:libeay32'
-$env:OPENSSL_LIB_DIR            = $ChocolateyHabitatLibDir
-$env:OPENSSL_INCLUDE_DIR        = $ChocolateyHabitatIncludeDir
-$env:LIBZMQ_PREFIX              = Split-Path $ChocolateyHabitatLibDir -Parent
+function Get-CargoCommand($toolchain) {
+    "cargo +$toolchain"
+}
 
-Write-RustToolVersion
+If($Features) {
+    $featureString = "--features $Features"
+} Else {
+    $featureString = ""
+}
 
-if(!(Test-Path "$env:TEMP\cacert.pem")) {
-    Write-Host "Downloading cacerts.pem"
-    $current_protocols = [Net.ServicePointManager]::SecurityProtocol
-    try {
-        [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
-        Invoke-WebRequest -UseBasicParsing -Uri "http://curl.haxx.se/ca/cacert.pem" -OutFile "$env:TEMP\cacert.pem"
-    }
-    finally {
-        [Net.ServicePointManager]::SecurityProtocol = $current_protocols
-    }
+# On buildkite, the rust binaries will be directly in C:
```

Comment:
Would it be worthwhile to add a specific Buildkite flag to the script?

I'm just wondering if this would misbehave if a user happened to have Rust installed like we do in Buildkite?